homeassistant/components/alarm_control_panel/device_condition.py

```diff
@@ -0,0 +1,162 @@
+                CONF_TYPE: CONDITION_DISARMED,
+            },
+            {
+                CONF_CONDITION: "device",
+                CONF_DEVICE_ID: device_id,
+                CONF_DOMAIN: DOMAIN,
+                CONF_ENTITY_ID: entry.entity_id,
+                CONF_TYPE: CONDITION_TRIGGERED,
+            },
+        ]
+        if supported_features & SUPPORT_ALARM_ARM_HOME:
+            conditions.append(
+                {
+                    CONF_CONDITION: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: CONDITION_ARMED_HOME,
+                }
+            )
+        if supported_features & SUPPORT_ALARM_ARM_AWAY:
+            conditions.append(
+                {
+                    CONF_CONDITION: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: CONDITION_ARMED_AWAY,
+                }
+            )
+        if supported_features & SUPPORT_ALARM_ARM_NIGHT:
+            conditions.append(
+                {
+                    CONF_CONDITION: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: CONDITION_ARMED_NIGHT,
+                }
+            )
+        if supported_features & SUPPORT_ALARM_ARM_CUSTOM_BYPASS:
+            conditions.append(
+                {
+                    CONF_CONDITION: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: CONDITION_ARMED_CUSTOM_BYPASS,
+                }
+            )
+
+    return conditions
+
+
+def async_condition_from_config(
+    config: ConfigType, config_validation: bool
+) -> condition.ConditionCheckerType:
+    """Create a function to test a device condition."""
+    if config_validation:
+        config = CONDITION_SCHEMA(config)
+    elif config[CONF_TYPE] == CONDITION_TRIGGERED:
+        state = STATE_ALARM_TRIGGERED
+    elif config[CONF_TYPE] == CONDITION_DISARMED:
+        state = STATE_ALARM_DISARMED
+    elif config[CONF_TYPE] == CONDITION_ARMED_HOME:
+        state = STATE_ALARM_ARMED_HOME
+    elif config[CONF_TYPE] == CONDITION_ARMED_AWAY:
+        state = STATE_ALARM_ARMED_AWAY
+    elif config[CONF_TYPE] == CONDITION_ARMED_NIGHT:
+        state = STATE_ALARM_ARMED_NIGHT
+    elif config[CONF_TYPE] == CONDITION_ARMED_CUSTOM_BYPASS:
+        state = STATE_ALARM_ARMED_CUSTOM_BYPASS
+
+    def test_is_state(hass: HomeAssistant, variables: TemplateVarsType) -> bool:
+        """Test if an entity is a certain state."""
+        return condition.state(hass, config[ATTR_ENTITY_ID], state)
```

Comment:
It looks like `state` will be undefined here if `config_validation` was true.

Reply:
Should I handle that like this?
https://github.com/home-assistant/core/blob/097b056324ce187722ff56cf34554860387d9d77/homeassistant/components/cover/device_condition.py#L174-L188

Reply:
Sorry, I'm not yet enough familiar with the background of this function to say.

@balloob?

Reply:
This will do: https://github.com/home-assistant/core/pull/29063/files#r393946959